pkg/kubernetes/overrides_test.go

```diff
@@ -276,19 +318,65 @@ func TestApplyDeploymentOverrides(t *testing.T) {
 				Spec: appsv1.DeploymentSpec{
 					Template: corev1.PodTemplateSpec{
 						Spec: corev1.PodSpec{
-							InitContainers: []corev1.Container{
+							Containers: []corev1.Container{
 								{
-									Name: "my-init",
-									Resources: corev1.ResourceRequirements{
-										Limits: corev1.ResourceList{
-											corev1.ResourceCPU: apimachineryresource.MustParse("50m"),
+									Name: "test",
+									LivenessProbe: &corev1.Probe{
+										ProbeHandler: corev1.ProbeHandler{
+											TCPSocket: &corev1.TCPSocketAction{
+												Port: intstr.FromString("rpc"),
+											},
 										},
 									},
 								},
 							},
+						},
+					},
+				},
+			},
+			override: &v1beta1.DeploymentOverride{
+				Spec: &v1beta1.DeploymentOverrideSpec{
+					Template: &v1beta1.PodTemplateSpecOverride{
+						Spec: &apiextensionsv1.JSON{
+							Raw: []byte(`{"containers":[{"name":"test","livenessProbe":{"$patch":"replace","tcpSocket":null,"exec":{"command":["echo","hi"]}}}]}`),
```

Comment:
You shouldn't need to specify "tcpSocket":null. Right?

Reply:
It turns out we have too. `replace` applies to every sibling keys, e.g. `tcpSocket`, `exec` in this case. `tcpSocket` will remain as is if not set explicitly to `null`. I also tried to replace `livenessProbe` but that seems removing other keys like delay seconds.